Generation IX Moves

Bassoonian opened the pull request:
Updates the PLA moves to SV standards and adds data for the new generation IX moves.

## Description
Some moves are marked as requiring a new flag that Lunos implemented as part of his gen 9 ability branch. They need to be uncommented to work. A bunch of moves are still missing custom effects because I'm not familiar enough with the battle codebase to do all of them myself and none of them have animations. Double Shock needs the correct string (it has a placeholder one because I couldn't find the one the game used and I didn't have time to test yet).

## **Discord contact info**
Jasper#5206

AsparagusEduardo commented on src/data/text/move_descriptions.h:
```diff
@@ -3042,13 +3042,200 @@ static const u8 sSandsearStormDescription[] = _(
+    "Its Attack and Speed are raised.");
+
+static const u8 sSnowscapeDescription[] = _(
+    "Summons a snowstorm that\n"
+    "strikes every turn.");
```
```suggestion
    "Summons a snowstorm that\n"
    "lasts for five turns.");
```

AsparagusEduardo commented on src/data/text/move_descriptions.h:
```diff
@@ -3042,13 +3042,200 @@ static const u8 sSandsearStormDescription[] = _(
+    "lowers the target's Attack.");
+
+static const u8 sHyperDrillDescription[] = _(
+    "A spinning pointed part pierces\n"
+    "through a foe's Protect.");
```
```suggestion
    "A spinning pointed part\n"
    "bypasses a foe's Protect.");
```

AsparagusEduardo commented on src/data/text/move_descriptions.h:
```diff
@@ -3042,13 +3042,200 @@ static const u8 sSandsearStormDescription[] = _(
+    "through a foe's Protect.");
+
+static const u8 sTwinBeamDescription[] = _(
+    "Mystical beams from the user's\n"
+    "eyes hit twice in a row.");
```
```suggestion
    "Mystical eye-beams that hit\n"
    "the target twice in a row.");
```

AsparagusEduardo commented on src/data/text/move_descriptions.h:
```diff
@@ -3042,13 +3042,200 @@ static const u8 sSandsearStormDescription[] = _(
+    "eyes hit twice in a row.");
+
+static const u8 sRageFistDescription[] = _(
+    "The more the user has been hit,\n"
+    "the stronger the move.");
```
```suggestion
    "The more the user has been\n"
    "hit, the stronger the move.");
```

AsparagusEduardo reviewed (changes requested):
I can't test in-battle effects as I haven't progressed much in the game yet, so I'll leave that part for @ghoulslash to review :eyes:

AsparagusEduardo commented on src/data/text/move_descriptions.h:
```diff
@@ -3042,13 +3042,200 @@ static const u8 sSandsearStormDescription[] = _(
+    "This attack never misses.");
+
+static const u8 sFlowerTrickDescription[] = _(
+    "A bouquet of flowers never misses\n"
+    "and always lands a critical hit.");
```
Updated suggestion, as I thought it was a bit confusing.

AsparagusEduardo commented on include/config.h:
```diff
@@ -70,6 +70,7 @@
 #define GEN_8 5
-#define GEN_LATEST GEN_8
+#define GEN_9 6
+#define GEN_LATEST GEN_9
```
Or at least until we have implemented enough significant changes to justify start having Gen 9 as the new standard.

AsparagusEduardo commented:
One more thing, remember to update `B_METRONOME_MOVES`.

Bassoonian commented on src/battle_tv.c:
```diff
@@ -299,7 +299,9 @@ static const u16 sPoints_MoveEffect[NUM_BATTLE_MOVE_EFFECTS] =
-    [EFFECT_CAMOUFLAGE] = 3
+    [EFFECT_CAMOUFLAGE] = 3,
+    [EFFECT_SLEEP_HIT] = 1,
+    [EFFECT_SPECIAL_ATTACK_UP_HIT] = 1,
```
We don't, but considering that the two new added things are equivalents of mechanics already present in Gen3, I included them out of analogy. I can remove them if you prefer

Bassoonian commented:
Updated flags and addressed Eduardo's concerns/comments. It's still a bit unclear which moves are forbidden by Metronome, Sleep Talk, Instruct etc. so that data still needs to be added in later once there's more clarity
